arch/arm/src/nrf53/nrf53_tim.c

```diff
@@ -0,0 +1,857 @@
+  .clear      = nrf53_tim_clear,
+  .configure  = nrf53_tim_configure,
+  .shorts     = nrf53_tim_shorts,
+  .count      = nrf53_tim_count,
+  .setcc      = nrf53_tim_setcc,
+  .getcc      = nrf53_tim_getcc,
+  .setpre     = nrf53_tim_setpre,
+  .setisr     = nrf53_tim_setisr,
+  .enableint  = nrf53_tim_enableint,
+  .disableint = nrf53_tim_disableint,
+  .checkint   = nrf53_tim_checkint,
+  .ackint     = nrf53_tim_ackint
+};
+
+#ifdef CONFIG_NRF53_TIMER0
+/* TIMER0 */
+
+struct nrf53_tim_priv_s g_nrf53_tim0_priv =
+{
+  .ops   = &nrf53_tim_ops,
+  .base  = NRF53_TIMER0_BASE,
+  .irq   = NRF53_IRQ_TIMER0,
+  .chan  = 4,
+  .inuse = false,
+};
+#endif
+
+#ifdef CONFIG_NRF53_TIMER1
+/* TIMER1 */
+
+struct nrf53_tim_priv_s g_nrf53_tim1_priv =
+{
+  .ops   = &nrf53_tim_ops,
+  .base  = NRF53_TIMER1_BASE,
+  .irq   = NRF53_IRQ_TIMER1,
+  .chan  = 4,
+  .inuse = false,
+};
+#endif
+
+#ifdef CONFIG_NRF53_TIMER2
+/* TIMER2 */
+
+struct nrf53_tim_priv_s g_nrf53_tim2_priv =
+{
+  .ops   = &nrf53_tim_ops,
+  .base  = NRF53_TIMER2_BASE,
+  .irq   = NRF53_IRQ_TIMER2,
+  .chan  = 4,
+  .inuse = false,
+};
+#endif
+
+#ifdef CONFIG_NRF53_TIMER3
+/* TIMER3 */
+
+struct nrf53_tim_priv_s g_nrf53_tim3_priv =
+{
+  .ops   = &nrf53_tim_ops,
+  .base  = NRF53_TIMER3_BASE,
+  .irq   = NRF53_IRQ_TIMER3,
+  .chan  = 6,
+  .inuse = false,
+};
+#endif
+
+#ifdef CONFIG_NRF53_TIMER4
+/* TIMER4 */
+
+struct nrf53_tim_priv_s g_nrf53_tim4_priv =
+{
+  .ops   = &nrf53_tim_ops,
+  .base  = NRF53_TIMER4_BASE,
+  .irq   = NRF53_IRQ_TIMER4,
+  .chan  = 6,
+  .inuse = false,
+};
+#endif
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nrf53_tim_getreg
+ *
+ * Description:
+ *   Get a 32-bit register value by offset
+ *
+ ****************************************************************************/
+
+static uint32_t nrf53_tim_getreg(struct nrf53_tim_dev_s *dev,
+                                 uint32_t offset)
+{
+  DEBUGASSERT(dev);
+
+  return getreg32(((struct nrf53_tim_priv_s *)dev)->base + offset);
+}
+
+/****************************************************************************
+ * Name: nrf53_tim_putreg
+ *
+ * Description:
+ *   Put a 32-bit register value by offset
+ *
+ ****************************************************************************/
+
+static void nrf53_tim_putreg(struct nrf53_tim_dev_s *dev,
+                             uint32_t offset,
+                             uint32_t value)
+{
+  DEBUGASSERT(dev);
+
+  putreg32(value, ((struct nrf53_tim_priv_s *)dev)->base + offset);
+}
+
+/****************************************************************************
+ * Name: nrf53_tim_irq2reg
+ *
+ * Description:
+ *   Get the value of the interrupt register corresponding to the given
+ *   interrupt source
+ *
```

Comment:
Please include the Input Parameters and the Return

Reply:
I prefer to keep it as it is for now, for compliance with NRF52

Reply:
Ok, hahaha